analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -121,9 +128,37 @@ public void RegisterNodeAction<TSyntaxKind>(GeneratedCodeRecognizer generatedCod
         // the decision is made on based on whether the project contains the analyzer as NuGet).
         if (context.HasMatchingScope(supportedDiagnostics)
             && context.ShouldAnalyzeTree(sourceTree, generatedCodeRecognizer)
+            && context.Options.ParseSonarLintXmlSettings() is { } sonarLintSettings
+            && IsIncluded(context.Compilation, PropertiesHelper.ReadSourceFileInclusionsProperty(sonarLintSettings, context.Compilation.Language), context.Tree.FilePath)
```

Comment:
I suggest to move this logic to `ShouldAnalyzeTree`

Reply:
Actually, ShouldAnalyzeTree is used in other locations (SonarReportingContextBase and WcfMissingContractAttribute rule) so moving the check inside that method would result in various failed UTs.

Reply:
After an offline conversation, we agreed that the suggestion to put the file path control inside ShouldAnalyzeTree was very appropriate. The other locations that are now calling ShouldAnalyzeTree need to check the file path as well, so everything needs to be part of that method. As a downside, we have some UTs failing, but since the implementation itself is not within the scope of the current sprint we will not spend time fixing them.